sql/sql_zip_dict.cc

```diff
@@ -20,27 +20,283 @@ this program; if not, write to the Free Software Foundation, Inc.,
 
 #include "sql/sql_zip_dict.h"
 
+#include <iostream>
+#include "sql/dd/impl/bootstrapper.h"
+#include "sql/dd/impl/transaction_impl.h"
+#include "sql/dd/types/column.h"               // dd::enum_column_types
+#include "sql/dd/types/column_type_element.h"  // dd::Column_type_element
+#include "sql/dd/types/table.h"                // dd::enum_column_types
 #include "sql/derror.h"
+#include "sql/field.h"
 #include "sql/handler.h"
+#include "sql/sql_base.h"
 #include "sql/sql_class.h"  // THD
 #include "sql/sql_table.h"  // write_bin_log
+#include "sql/table.h"
+#include "sql/thd_raii.h"
+#include "sql/transaction.h"
+
+namespace compression_dict {
+
+/** Fields of mysql.compression_dictionary table */
+static constexpr uint32_t COMPRESSION_DICTIONARY_FIELD_ID = 0;
+static constexpr uint32_t COMPRESSION_DICTIONARY_FIELD_VERSION = 1;
+static constexpr uint32_t COMPRESSION_DICTIONARY_FIELD_NAME = 2;
+static constexpr uint32_t COMPRESSION_DICTIONARY_FIELD_DATA = 3;
+
+/** Index of compression_dictionary table */
+static constexpr uint32_t COMPRESSION_DICTIONARY_NAME_INDEX = 1;
+static constexpr uint32_t COMPRESSION_DICTIONARY_VERSION = 1;
+
+static const char *COMPRESSION_DICTIONARY_DB = "mysql";
+static const char *COMPRESSION_DICTIONARY_TABLE = "compression_dictionary";
+
+/** Fields of mysql.compression_dictionary_cols table */
+static constexpr uint32_t COMPRESSION_DICTIONARY_COLS_FIELD_TABLE_ID = 0;
+static constexpr uint32_t COMPRESSION_DICTIONARY_COLS_FIELD_COLUMN_ID = 1;
+static constexpr uint32_t COMPRESSION_DICTIONARY_COLS_FIELD_DICT_ID = 2;
+
+/** Index of compression_dictionary_cols table */
+static constexpr uint32_t COMPRESSION_DICTIONARY_COLS_TABLE_INDEX = 0;
+
+static const char *COMPRESSION_DICTIONARY_COLS_DB = "mysql";
+static const char *COMPRESSION_DICTIONARY_COLS_TABLE =
+    "compression_dictionary_cols";
+
+/** Max window size (2^15) minus 262 */
+static constexpr uint32_t MAX_DATA_LENGTH = 32506;
+
+/** Schema definition of mysql.compression_dictionary */
+const dd::String_type dict_table_str =
```

Comment:
These do not seem to be exported?
```suggestion
static const dd::String_type dict_table_str =
```
etc